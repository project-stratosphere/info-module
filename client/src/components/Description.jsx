import PropTypes from 'prop-types';
import React from 'react';
import {
  Wrapper,
  ShortDescription,
  MoreDescription,
  DescriptionContainer,
  DescriptionHead,
  DescriptionBody,
  Clickable,
} from './styles/Description.styles';

export default class Description extends React.Component {
  constructor(props) {
    super(props);
    this.toggleDisplay = this.toggleDisplay.bind(this);
    this.state = {
      hidden: true,
    };
  }

  toggleDisplay() {
    this.setState({
      hidden: !this.state.hidden,
    });
  }

  render() {
    return (
      <Wrapper>
        <ShortDescription>
          {this.props.short_description}
        </ShortDescription>
        <Clickable
          onClick={this.toggleDisplay}
          hidden={!this.state.hidden}
          className="read-more"
        >
          Read more about the space &#9660;
        </Clickable>
        <Clickable
          onClick={this.toggleDisplay}
          hidden={this.state.hidden}
          className="hide"
        >
          Hide &#9650;
        </Clickable>
        <MoreDescription
          hidden={this.state.hidden}
        >
          {
            this.props.more_description.map(description => (
              <DescriptionContainer
                key={description.head}
              >
                <DescriptionHead>
                  {description.head}
                </DescriptionHead>
                <DescriptionBody>
                  {description.body}
                </DescriptionBody>
              </DescriptionContainer>
            ))
          }
        </MoreDescription>
      </Wrapper>
    );
  }
}

Description.propTypes = {
  short_description: PropTypes.string,
  more_description: PropTypes.arrayOf(PropTypes.shape({
    head: PropTypes.string,
    body: PropTypes.string,
  })),
};

Description.defaultProps = {
  short_description: 'Short Description',
  more_description: [
    {
      head: 'String',
      body: 'String',
    },
  ],
};
