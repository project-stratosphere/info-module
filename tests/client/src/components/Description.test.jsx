import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Description from '../../../../client/src/components/Description';

Enzyme.configure({ adapter: new Adapter() });

const moreDescription = [
  {
    head: 'Mock Head',
    body: 'Mock Body',
  },
];

describe('Description', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Description
      short_description="Mock Short Description"
      more_description={moreDescription}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should change state when read-more and hide are clicked', () => {
    const output = Enzyme.shallow(<Description
      short_description="Mock Short Description"
      more_description={moreDescription}
    />);
    expect(output.state().hidden).toEqual(true);
    output.find('.read-more').simulate('click');
    expect(output.state().hidden).toEqual(false);
    output.find('.hide').simulate('click');
    expect(output.state().hidden).toEqual(true);
  });
});
