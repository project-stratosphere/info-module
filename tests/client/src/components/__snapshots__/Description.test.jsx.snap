// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly 1`] = `
<styled.div>
  <styled.div>
    Mock Short Description
  </styled.div>
  <styled.div
    className="read-more"
    hidden={false}
    onClick={[Function]}
  >
    Read more about the space ▼
  </styled.div>
  <styled.div
    className="hide"
    hidden={true}
    onClick={[Function]}
  >
    Hide ▲
  </styled.div>
  <styled.div
    hidden={true}
  >
    <styled.div
      key="Mock Head"
    >
      <styled.div>
        Mock Head
      </styled.div>
      <styled.div>
        Mock Body
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
